cmd/repo_add_test.go

```diff
@@ -106,8 +107,7 @@ func TestRepoAddErrors(t *testing.T) {
 			sandbox, cleanup := cmdtest.TestSetupWithSandbox(t, true)
 			defer cleanup()
 
-			// see how many repos we currently have
-			startRepos := getRepoListOutput(t, sandbox)
```

Comment:
why delete this and the `endRepos` check below?

Reply:
When executing the `Badly formatted repo config`, if the startRepo & endRepo check lines are included the test fails because it can't read the config file (only ever the `bad_format_repository_config` test run - the other invalid tests pass as expected). The failure it outputs is:
```
--- FAIL: TestRepoAddErrors (0.00s)
    --- FAIL: TestRepoAddErrors/Badly_formatted_repo_config (1.95s)
        /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testutils.go:103: Created testing project dir:  /private/var/folders/39/j9wy27654gvfd2_jtr2n81980000gn/T/appsody-TestRepoAddErrors-Badly_formatted_repo_config-160842844/testrepoadderrors-badly_formatted_repo_config-160842844
        /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testutils.go:110: Created testing project dir:  /private/var/folders/39/j9wy27654gvfd2_jtr2n81980000gn/T/appsody-TestRepoAddErrors-Badly_formatted_repo_config-160842844/testrepoadderrors-badly_formatted_repo_config-160842844
        /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testutils.go:111: Created testing config dir:  /private/var/folders/39/j9wy27654gvfd2_jtr2n81980000gn/T/appsody-TestRepoAddErrors-Badly_formatted_repo_config-160842844/config
        /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testutils.go:134: Directory copy of /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testdata to /private/var/folders/39/j9wy27654gvfd2_jtr2n81980000gn/T/appsody-TestRepoAddErrors-Badly_formatted_repo_config-160842844 was successful 
        /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testutils.go:135: 
        /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testutils.go:216: Running appsody in the test sandbox with args:  [repo list -v --config /private/var/folders/39/j9wy27654gvfd2_jtr2n81980000gn/T/appsody-TestRepoAddErrors-Badly_formatted_repo_config-160842844/testdata/bad_format_repository_config/config.yaml]
        /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testutils.go:211: [Debug] Logging to file /Users/lukeweston/.appsody/logs/appsody2020-01-15T10-47-01.335004.log
        /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testutils.go:211: [Debug] Updating reference to 'index.docker.io' in /private/var/folders/39/j9wy27654gvfd2_jtr2n81980000gn/T/appsody-TestRepoAddErrors-Badly_formatted_repo_config-160842844/testdata/bad_format_repository_config/config.yaml
        /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testutils.go:211: [Debug] Writing config file /private/var/folders/39/j9wy27654gvfd2_jtr2n81980000gn/T/appsody-TestRepoAddErrors-Badly_formatted_repo_config-160842844/testdata/bad_format_repository_config/.appsody.yaml
        /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testutils.go:211: [Debug] Getting latest version https://github.com/appsody/appsody/releases/latest
        /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testutils.go:211: [Debug] Got version 0.5.4
        /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testutils.go:211: [Warning] Unable to read repository file
        /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testutils.go:211: [Debug] Running with command line args: appsody repo list -v --config /private/var/folders/39/j9wy27654gvfd2_jtr2n81980000gn/T/appsody-TestRepoAddErrors-Badly_formatted_repo_config-160842844/testdata/bad_format_repository_config/config.yaml
        /Users/lukeweston/go/src/github.com/appsody/appsody/cmd/testutils.go:211: [Error] Failed to parse repository file yaml: line 5: mapping values are not allowed in this context
```
Initially I put in a `sandbox.SetConfigInTestData("default_repository_config")` before and after both the `startRepo` and `endRepo` checks, meaning it only ran with `bad_format_repository_config`  when executing `repo add` - but then I deleted it altogether (to prioritise making the tests run concurrently) with the intention of asking what would be the best resolution. I wasn't entirely sure if the check was completely necessary because all of the tests are expected negatives, and if they returned a non-nil value the tests would fail anyway; I do understand that it's to check that failed results don't still add unintentional repo's, though.

Reply:
The startRepos and endRepos check is an important part of these tests. It make sure that nothing was changed in the repo list when we run bad commands.

I wonder how the bad_format_repository check was working before? I think we should try it again with the start and end repos and see if we can get it working.